File: src/Common/Models/Account.java
package Common.Models;

import Common.Models.Base.*;

public class Account extends ReferenceChildEntity<AccountSubGroup> {

    private Currency currency;
    private Correspondent correspondent;
    private Category category;
    private Project project;

    public Currency getCurrency() {
        return currency;
    }
    public void setCurrency(Currency currency) {
        this.currency = currency;
    }

    public Correspondent getCorrespondent() {
        return correspondent;
    }
    public void setCorrespondent(Correspondent correspondent) {
        this.correspondent = correspondent;
    }

    public Category getCategory() {
        return category;
    }
    public void setCategory(Category category) {
        this.category = category;
    }

    public Project getProject() {
       return project;
    }
    public void setProject(Project project) {
        this.project = project;
    }
}

File: backend/Business.Models/Entities/Account.cs
using Business.Models.Entities.Base;

namespace Business.Models.Entities;

public class Account : ElementEntity<AccountGroup, Account>
{
	public Currency Currency { get; set; } = null!;
	public Guid CurrencyId { get; set; }
	public Category? Category { get; set; }
	public Guid? CategoryId { get; set; }
	public Correspondent? Correspondent { get; set; }
	public Guid? CorrespondentId { get; set; }
	public Project? Project { get; set; }
	public Guid? ProjectId { get; set; }
}
